src/x11/x.c

```diff
@@ -894,7 +894,11 @@ gboolean x_mainloop_fd_dispatch(GSource *source, GSourceFunc callback, gpointer
         unsigned int state;
         while (XPending(xctx.dpy) > 0) {
                 XNextEvent(xctx.dpy, &ev);
-                LOG_D("Processing event: %s\n", xevent_type_to_string(ev.type));
+                const char *typename = xevent_type_to_string(ev.type);
+                if (typename)
+                        LOG_D("XEvent: processing '%s'", typename);
```

Comment:
You can avoid having to use if logic for a simple debug log (which I dislike) by changing the statement to
```
LOG_D("XEvent: processing '%d: %s'", ev.type, typename);
```
and update `xevent_type_to_string` to return "Unknown" on an unkown event.